src/controller/sql/mod.rs

```diff
@@ -526,30 +526,42 @@ impl SqlIncorporator {
         let universe = mig.universe();
         // no QG-level reuse possible, so we'll build a new query.
         // first, compute the MIR representation of the SQL query
-        let mut mir = self.mir_converter.named_query_to_mir(
+        let (sec, og_mir, table_mapping, base_name) = self.mir_converter.named_query_to_mir(
             query_name,
             query,
             &qg,
             is_leaf,
             universe.clone(),
         );
 
-        trace!(self.log, "Unoptimized MIR:\n{}", mir.to_graphviz().unwrap());
+        trace!(self.log, "Unoptimized MIR:\n{}", og_mir.to_graphviz().unwrap());
 
         // run MIR-level optimizations
-        mir = mir.optimize();
+        let mut mir = og_mir.optimize(table_mapping.clone(), sec);
 
         trace!(self.log, "Optimized MIR:\n{}", mir.to_graphviz().unwrap());
 
+        if sec {
+            match table_mapping.clone() {
```

Comment:
No clone needed here, I think?  `table_mapping` isn't used again, so we should be able to take ownership